aggregator/leader_flush_mgr.go

```diff
@@ -282,20 +322,43 @@ func (t *leaderFlushTask) Run() {
 		go mgr.persistFlushTimes(t.flushTimes)
 	}
 
-	if t.shouldFlush {
-		var wgWorkers sync.WaitGroup
-		start := mgr.nowFn()
-		for _, flusher := range t.flushers {
-			flusher := flusher
-			wgWorkers.Add(1)
-			mgr.workers.Go(func() {
-				flusher.Flush()
-				wgWorkers.Done()
-			})
+	if !t.shouldFlush {
+		return
+	}
+
+	// If the instance is not found in the placement, this means the instance is no longer
+	// considered as part of the aggregation cluster and as such we should stop flushing.
+	shards, err := mgr.ownedShards()
+	if err == errInstanceNotFoundInPlacement {
+		return
+	}
+
+	var (
+		wgWorkers sync.WaitGroup
+		start     = mgr.nowFn()
+	)
+	for _, flusher := range t.flushers {
+		// By default traffic is cut off from a shard, unless the shard is in the list of
+		// shards owned by the instance, in which case the cutover time and the cutoff time
+		// are set to the corresponding cutover and cutoff times of the shard.
+		var cutoverNanos, cutoffNanos int64
+		if shards != nil {
+			shardID := flusher.Shard()
+			if shard, found := shards.Shard(shardID); found {
+				cutoverNanos = shard.CutoverNanos()
+				cutoffNanos = shard.CutoffNanos()
+			}
```

Comment:
Isn't something bad going to happen if the shard is not found? Can you still flush with cutOver and cutOff set to 0?

Reply:
Not necessarily. This may happen if the current instance no longer owns this shard (i.e., the shard is removed from the instance's shard list in the placement) and we are in the middle of asynchronously closing the removed shards (which removes the flusher from the list of flushers) but aren't done yet. Once the shards are closed, it should no longer hit this part again.